src/position-manager/PositionConfigPositionManager.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: UNLICENSED
+// Copyright (c) 2025 Aave Labs
+pragma solidity 0.8.28;
+
+import {ISpoke} from 'src/spoke/interfaces/ISpoke.sol';
+import {ConfigPermissionsMap} from 'src/position-manager/libraries/ConfigPermissionsMap.sol';
+import {PositionManagerBase} from 'src/position-manager/PositionManagerBase.sol';
+import {IPositionConfigPositionManager, ConfigPermissions} from 'src/position-manager/interfaces/IPositionConfigPositionManager.sol';
+
+/// @title PositionConfigPositionManager
+/// @author Aave Labs
+/// @notice Position manager to handle position configuration actions on behalf of users.
+contract PositionConfigPositionManager is IPositionConfigPositionManager, PositionManagerBase {
+  using ConfigPermissionsMap for ConfigPermissions;
+
+  mapping(address owner => mapping(address caller => ConfigPermissions)) private _configPermissions;
+
+  /// @dev Constructor.
+  /// @param spoke_ The address of the spoke contract.
+  constructor(address spoke_) PositionManagerBase(spoke_) {}
+
+  /// @inheritdoc IPositionConfigPositionManager
+  function setGlobalPermission(address caller, bool permission) external {
+    _configPermissions[msg.sender][caller] = _configPermissions[msg.sender][caller]
+      .setFullPermissions(permission);
+
+    emit GlobalConfigPermissionUpdated(msg.sender, caller, permission);
```

Comment:
we can just have one event which signals the new permissions bitmap i think, we don't need separate events for each type of action

Reply:
ffa90f049ebaea40a9f75f9d63978945e65bd2e0